app.js

```diff
@@ -61,16 +61,24 @@ if (opts.debug) logger.level = 'debug';
 
 // AP: Detect if we're running under Windows Subsystem for Linux. Temporary modification
 // of process.env is allowed: https://nodejs.org/api/process.html#process_process_env
-if ((child_process.execSync('uname -a').toString().indexOf('Microsoft') > -1) && (opts.wsl))
+if (child_process.execSync('uname -a').toString().indexOf('Microsoft') > -1)
     process.env.wsl = true;
 
-// AP: Allow setting of tmpDir (used in lib/base-compiler.js & lib/exec.js) through
-// opts. WSL requires a tmpDir as it can't see Linux volumes so set default to c:\tmp.
-if (opts.tmpDir)
+// AP: Allow setting of tmpDir (used in lib/base-compiler.js & lib/exec.js) through opts.
+// WSL requires a directory on a Windows volume. Set that to Windows %TEMP% if no tmpDir supplied.
+// If a tempDir is supplied then assume that it will work for WSL processes as well. 
+if (opts.tmpDir) {
     process.env.tmpDir = opts.tmpDir;
-else if (process.env.wsl)
-    process.env.tmpDir = "/mnt/c/tmp";
-
+    process.env.winTmp = opts.tmpDir;
+}
+else if (process.env.wsl) {
+    // Dec 2017 preview builds of WSL include /bin/wslpath; do the parsing work for now.
+    // Parsing example %TEMP% is C:\Users\apardoe\AppData\Local\Temp
+    var windowsTemp = child_process.execSync('cmd.exe /c echo %TEMP%').toString().replace(/\\/g, "/");
+    var driveLetter = windowsTemp.substring(0, 1).toLowerCase();
+    var directoryPath = windowsTemp.substring(2).trim();
+    process.env.winTmp = "/mnt/".concat(driveLetter).concat(directoryPath);
```

Comment:
Just as a minor note, `path.join()` is meant to be used for paths concatenation, but I don't know if it would work for this setup (I'm guessing it does, but you never know!)

Reply:
Yes, path.join should work here. I don't know that path.split would work as it's running on Linux and getting back a Windows path. Let me look at this.

Reply:
If you want to do win32 path operations and be sure you are getting that result back use path.win32

https://nodejs.org/api/path.html#path_path_win32

Reply:
This would be preferable to string operations.

Reply:
@jaredwy, thank you for the link, but I don't see how there are path.win32 operations that will do this for me. There's a path.win32 function that will return a drive letter as `C:\` but I'd still have to `substring` and `tolower` it in order to turn it into `/mnt/c/`.